src/Simplex/Messaging/Agent.hs

```diff
@@ -1862,6 +1871,7 @@ cleanupManager c@AgentClient {subQ} = do
     deleteConns =
       withLock (deleteLock c) "cleanupManager" $ do
         void $ withStore' c getDeletedConnIds >>= deleteDeletedConns c
+        void $ withStore' c getDeletedWaitingDeliveryConnIds >>= deleteDeletedWaitingDeliveryConns c
```

Comment:
perhaps better to refactor deleteConnections_ so that connections are read outside of prepareDeleteConnections_ and passed to it, and call only deleteConnQueues here so that it can notify with DEL_CONN (or create a different function?)